src/dist/replication/lib/replica.cpp

```diff
@@ -197,13 +225,11 @@ void replica::check_state_completeness()
 
     auto mind = _stub->_log->max_gced_decree(get_gpid(), _app->init_info().init_offset_in_shared_log);
     dassert(mind <= last_durable_decree(), "");
-    _stub->_log->check_valid_start_offset(get_gpid(), _app->init_info().init_offset_in_shared_log);
```

Comment:
check_valid_start_offset() is already invoked in max_gced_decree(), so
no need to invoke twice


On Thu, Jun 16, 2016 at 11:52 AM, Zhenyu Guo <notifications@github.com>
wrote:

> In src/dist/replication/lib/replica.cpp
> <https://github.com/imzhenyu/rDSN/pull/403#discussion_r67283870>:
>
> > @@ -197,13 +225,11 @@ void replica::check_state_completeness()
> >
> >      auto mind = _stub->_log->max_gced_decree(get_gpid(), _app->init_info().init_offset_in_shared_log);
> >      dassert(mind <= last_durable_decree(), "");
> > -    _stub->_log->check_valid_start_offset(get_gpid(), _app->init_info().init_offset_in_shared_log);
>
> why the checkings are removed?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/403/files/89a41d5e513a99014291d1ad76187ae04459b1ab#r67283870>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAgQPe5fW7DlvXvoFvxaKJ8Ir6l3swWyks5qMMhugaJpZM4I2J8D>
> .
>